src/datasources/alerts-api/tenderly-api.service.spec.ts

```diff
@@ -100,14 +100,13 @@ describe('TenderlyApi', () => {
   });
 
   it('should forward error', async () => {
-    mockNetworkService.post.mockImplementation(() =>
-      Promise.reject('Unexpected error'),
-    );
-    mockHttpErrorFactory.from.mockImplementation(
-      () => new DataSourceError('Unexpected error'),
+    const expected = new DataSourceError('Unexpected error');
+    mockHttpErrorFactory.from.mockReturnValue(expected);
```

Comment:
Not specifically related with this PR but would affect the changes proposed: why not use the real instance of the `HttpErrorFactory`?

That way we cover more of the actual implementation and we avoid having to set the correct mock value set here.

Reply:
I've adjusted it in 193e5cc012a581c84310250f9ebb28e0567a3ced but as per [your suggestion](https://github.com/safe-global/safe-client-gateway/pull/801#discussion_r1383071213), maybe this will need to be further changed.

Reply:
Yes my sugggestion was to remove the Mock entirely and use the real implementation:

```typescript
it('should forward error', async () => {
  const status = faker.internet.httpStatusCode({ types: ['serverError'] });
  const error = {
    status,
    data: {
      message: 'Unexpected error',
    },
  };
  mockNetworkService.post.mockRejectedValueOnce(error);

  await expect(service.addContracts([])).rejects.toThrowError(
    new DataSourceError('Unexpected error', status),
  );

  expect(mockNetworkService.post).toHaveBeenCalledTimes(1);
});
```

In this example, the only component being mocked is the HTTP component as it's more complex and we want to abstract it away for this test.

This requires you changing the way `TenderlyApi` is constructed:

```typescript
service = new TenderlyApi(
  fakeConfigurationService,
  mockNetworkService,
  new HttpErrorFactory(),
);
```

Reply:
I see, my bad! I have changed to using a real implementation in 1b7d94a3cade2cdf3d86e5c12f982c2099259eec.